### PR TITLE
Add client configuration for more retries; specify standard mode

### PR DESCRIPTION
**Pull request by tthyer**
Attempt to fix SC-179 by adding more than the 3 standard retries; also, updating from `legacy` to `standard` mode for retries.

Adding more retries will increase the likelihood that the request will succeed since boto3 is using [exponential backoff](https://github.com/boto/botocore/blob/1.16.25/botocore/retryhandler.py#L39) for retries, with a limit of 20 seconds between retries.

There is a danger with more retries that we can be rate limited; if that happens, we can try the experimental `adaptive` mode; but it is unlikely to happen with exponential backoff.

If there are questions, the [Boto3 Retries guide](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/retries.html) is excellent.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ pre-commit = "*"
 
 [packages]
 aws-sam-cli = "*"
-boto3 = "*"
+boto3 = ">=1.13"
```
Comment (tthyer) [inline Pipfile:14]: Specified greater or equal here to pick up recent mode changes in boto3.

**Comment by tthyer** [comment]
@zaro0508's recommendation -- to avoid rate limiting, switching to adaptive mode. We will also use `DependsOn` in the templates to avoid resource contention with the other S3 lambda that is commonly used.

**Comment by zaro0508** [comment]
I'm not opposed to this change although I don't think it's necessary.  I think the contention is from how we are using this custom resource.  https://github.com/Sage-Bionetworks/service-catalog-library/blob/master/s3/sc-s3-encrypted-ra-v1.0.0.yaml#L40

The contention is more likely that both of your custom resources, S3BucketTagger and S3BucketPolicy are running in parallel.  I would try orchestrating those CRs with a CFN `DependsOn` property to see if it fixes the problem first.   

As you pointed out, If we make this change we run a higher risk of getting rate limitted by AWS.

**Comment by tthyer** [comment]
@zaro0508 I will close this PR and try using just DependsOn. We can re-open this if that simpler solution doesn't work.

**Pull request closed without merging**